### PR TITLE
Use string format for extension quoting

### LINKED ISSUE #603: Failed to install extension with hyphenated name
### Cookbook version
git master (since 7.1.3 was released *right* before #595 was merged)

### Chef-client version
14.11.21

### Platform Details
CentOS 7.6

### Scenario:
Install the `uuid-ossp` extension with `postgresql_extension`

### Steps to Reproduce:
```rb
postgresql_extension 'uuid-ossp' do
  database dbname
end
```

### Expected Result:
Success

### Actual Result:
<details>
  <summary>Error log</summary>

```
==> default:   * postgresql_extension[uuid-ossp] action create
==> default:
==> default:     * bash[CREATE EXTENSION uuid-ossp] action run
==> default:
==> default:       [execute] ERROR:  syntax error at or near "-"
==> default:                 LINE 1: CREATE EXTENSION IF NOT EXISTS uuid-ossp
==> default:                                                            ^
==> default:
==> default:
==> default: ================================================================================
==> default:       Error executing action `run` on resource 'bash[CREATE EXTENSION uuid-ossp]'
==> default:       ================================================================================
==> default:
==> default:
==> default:       Mixlib::ShellOut::ShellCommandFailed
==> default:       ------------------------------------
==> default:
==> default:
==> default: Expected process to exit with [0], but received '1'
==> default:       ---- Begin output of "bash"  "/tmp/chef-script20190318-9737-1glk6rq" ----
==> default:
==> default: STDOUT:
==> default:       STDERR: ERROR:  syntax error at or near "-"
==> default:
==> default:       LINE 1: CREATE EXTENSION IF NOT EXISTS uuid-ossp
==> default:                                                  ^
==> default:
==> default:       ---- End output of "bash"  "/tmp/chef-script20190318-9737-1glk6rq" ----
==> default:       Ran "bash"  "/tmp/chef-script20190318-9737-1glk6rq" returned 1
==> default:
==> default:
==> default:
==> default:       Resource Declaration:
==> default:       ---------------------
==> default:       # In /var/chef/cache/cookbooks/postgresql/resources/extension.rb
==> default:
==> default:
==> default:
==> default:  29:   bash "CREATE EXTENSION #{new_resource.name}" do
==> default:        30:     code create_extension_sql(new_resource)
==> default:        31:     user 'postgres'
==> default:        32:     action :run
==> default:        33:     not_if { follower? || extension_installed?(new_resource) }
==> default:        34:   end
==> default:
==> default:  35: end
==> default:
==> default:
==> default:       Compiled Resource:
==> default:
==> default: ------------------
==> default:
==> default: # Declared in /var/chef/cache/cookbooks/postgresql/resources/extension.rb:29:in `block in class_from_file'
==> default:
==> default:
==> default:
==> default: bash("CREATE EXTENSION uuid-ossp") do
==> default:         action [:run]
==> default:         default_guard_interpreter :default
==> default:         command nil
==> default:
==> default:   backup 5
==> default:         interpreter "bash"
==> default:         declared_type :bash
==> default:
==> default:   cookbook_name "dev"
==> default:         code "/usr/bin/psql -c \"CREATE EXTENSION IF NOT EXISTS uuid-ossp\" -d portal -U postgres --port 5432"
==> default:         domain nil
==> default:         user "postgres"
==> default:
==> default:         not_if { #code block }
==> default:       end
==> default:
==> default:
==> default:
==> default:       System Info:
==> default:       ------------
==> default:       chef_version=14.11.21
==> default:       platform=centos
==> default:       platform_version=7.6.1810
==> default:
==> default: ruby=ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
==> default:       program_name=/bin/chef-solo
==> default:       executable=/opt/chef/bin/chef-solo
```

</details>

I imagine the issue is due to the extension name not being quoted [here](https://github.com/sous-chefs/postgresql/blob/master/libraries/helpers.rb#L84).

### PR DESCRIPTION
**Pull request by JanStevens**
# Description

This allows `'"uuid-ossp"' to keep the double quotes so it becomes `CREATE EXTENSION IF NOT EXISTS "uuid-ossp"` but also allows to extend on the SQL for example `CREATE EXTENSION IF NOT EXISTS citext SCHEMA pg_catalog`

## Issues Resolved

#603 

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JanStevens** [comment]
I tried multiple things but I'm not getting to a uniform solution where both `"` and `'` could freely be used. My last change cause all other tests to fail.

What could be a better solution is writing the query to file and then execute it with psql. That way there are no quoting issues.

**Comment by JanStevens** [comment]
And temp files are a bad idea across methods, it's possible that they are already deleted because they are garbage collected

File: libraries/helpers.rb [hunk 1]
```diff
@@ -20,9 +20,11 @@ module Helpers
     include Chef::Mixin::ShellOut
 
     require 'securerandom'
+    require 'tempfile'
 
     def psql_command_string(new_resource, query, grep_for: nil, value_only: false)
-      cmd = "/usr/bin/psql -c \"#{query}\""
+      query_file = Tempfile.create('sql-query') { |f| f << query }
+      cmd = "/usr/bin/psql -f #{query_file.path}"
```
Comment (damacus) [inline libraries/helpers.rb]: How come we need the Tempfile here?

File: metadata.rb [hunk 2]
```diff
@@ -4,7 +4,6 @@
 maintainer_email  'help@sous-chefs.org'
 license           'Apache-2.0'
 description       'Installs and configures postgresql for clients or servers'
-long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
```
Comment (damacus) [inline metadata.rb]: 💥

File: resources/access.rb [hunk 3]
```diff
@@ -16,12 +16,12 @@
 # limitations under the License.
 #
 
-property :access_type,   String, required: true, default: 'local'
```
Comment (damacus) [inline resources/access.rb]: Ho come we're getting rid of required here?

File: libraries/helpers.rb (hunk 1 above)
Comment (JanStevens) [reply libraries/helpers.rb]: It is an attempt at getting around quoting issues when you try to install an extension with a hypen in it like `uuid-ossp`. Sadley it didn't work at all

File: resources/access.rb (hunk 3 above)
Comment (JanStevens) [reply resources/access.rb]: I got those errors, default and required are mutually exclusive ref: https://app.circleci.com/jobs/github/sous-chefs/postgresql/1226

File: metadata.rb (hunk 2 above)
Comment (JanStevens) [reply metadata.rb]: ref: https://app.circleci.com/jobs/github/sous-chefs/postgresql/1226

File: libraries/helpers.rb (hunk 1 above)
Comment (damacus) [reply libraries/helpers.rb]: `%w()` does seem like the right direction in this case

**Pull request merged**